### PR TITLE
DBT-762: Dbt-core version upgrade to v1.5

### PR DESCRIPTION
**Pull request by nsharma-25**
## Describe your changes
Updating dbt-core version to v1.5
## Internal Jira ticket number or external issue link
https://jira.cloudera.com/browse/DBT-762
## Testing procedure/screenshots(if appropriate):
https://gist.github.com/nsharma-25/b55aae879240af016ceee67c3c2a8105
pip list command output- https://gist.github.com/nsharma-25/d0e74be41dfea097514a488145a1574a
pipdeptree command output- https://gist.github.com/nsharma-25/905e031304a3ba5da10d6ec7a979bf2c
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have formatted my added/modified code to follow pep-8 standards
- [x] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

File: dbt/adapters/hive/connections.py [hunk 1]
```diff
@@ -99,6 +99,10 @@ def __post_init__(self):
     def type(self):
         return "hive"
 
+    @property
+    def unique_field(self) -> str:
```
Comment (niteshy) [inline dbt/adapters/hive/connections.py:103]: Why do we need this change? Can you elaborate on the changes related to dbt-core 1.5 ?

Comment (nsharma-25) [reply dbt/adapters/hive/connections.py:103]: > Why do we need this change? Can you elaborate on the changes related to dbt-core 1.5 ?

Without adding this implementation, I was getting the below error:
```
NotImplementedError: unique_field not implemented for the base credentials class
```

**Review by niteshy — CHANGES_REQUESTED** [review]
@nsharma-25 @vamshikolanu not even simple tests are working with changes. https://gist.github.com/niteshy/cb354c4242fffd3335d154fe8e722141#file-error-logs-L83-L96

**Review by niteshy — CHANGES_REQUESTED** [review]
close this please

**Pull request closed without merging**